### ISSUE TITLE
Support for parameter Type list

### ISSUE THREAD

**Issue by simonnagl** [buschmais/jqa-core-framework#18]
I implemented some constraints which I want to apply to different projects using a custom plugin.
For different projects I need to exclude some special files which are allowed to violate the constraint. For this special cases I tried to use arguments. Here is an example:

```xml
    <constraint id="tm20:UppercaseLiquibaseChangesetTableNames">
        <requiresParameter name="exceptFileNames" type="String" defaultValue=['/com/bmw/tmtool/liquibase/base/2017-05-10_TM20-21.xml']/>
        <description>Find Liquibase changesets with not uppercase table names</description>
        <cypher><![CDATA[
            MATCH (doc:Xml:Document), // Here comes the path(s) to check
             WHERE  // Here comes the constraint violation part
             // exclude special cases
            AND none(fileName IN {exceptFileNames} WHERE doc.fileName = fileName)
            RETURN doc.fileName as file
        ]]></cypher>
    </constraint>
```

It seems this is currently not supportet as `type="String"` puts `''` around the value during substitutions. Is there a solution for my use-case currently?
I am willing to help to implement this issue. Are there any plans on how to implement this?

Thank you for your time and answer.

**Comment by DirkMahler** [comment]
Array parameters are not supported yet but in your case you can use a string parameter consisting of comma separated values that are splitted by your cypher rule, e.g.

`WITH
  split({expectFileNames},  ",") as fileNames
MATCH
  ...
WHERE
  ...
  (fileName IN fileNames)
...'

**Comment by simonnagl** [comment]
Thank you for the tip, this is enough for know. A parameter type for this would be great anyways. Should I leave this open until it is implemented?

**Comment by obfischer** [comment]
Yes, please. We will talk about it after the next release.

**Comment by 62mkv** [comment]
..and when is the next release ?

**Comment by obfischer** [comment]
The latest unofficial release is 1.7.0 Milestone 3. (https://search.maven.org/search?q=g:com.buschmais.jqassistant)

**Comment by 62mkv** [comment]
yes, so may be milestone 1.5.0 on this issue should be ... updated somehow ?

**Comment by 62mkv** [comment]
or has it been resolved since then ?

**Comment by 62mkv** [comment]
looking at the `com.buschmais.jqassistant.core.analysis.api.rule.Parameter.Type` enum, seems that only supported types for parameters in current 1.7.0-SNAPSHOT version are: 
        CHAR,
        BYTE,
        SHORT,
        INT,
        LONG,
        FLOAT,
        DOUBLE,
        BOOLEAN,
        STRING

so, unfortunately not a straight-forward way to have lists, maps, etc as rule parameters. But, splitting string is not so bad actually! Thanks for suggestion!
